src/feature-extraction/lidar_sensor.jl

```diff
@@ -28,7 +28,7 @@ function LidarSensor(nbeams::Int;
     LidarSensor(angles, ranges, range_rates, max_range, ConvexPolygon(4))
 end
 nbeams(lidar::LidarSensor) = length(lidar.angles)
-function observe!(lidar::LidarSensor, scene::Scene, roadway::Roadway, vehicle_index::Int)
+function observe!(lidar::LidarSensor, scene::Frame{Entity{VehicleState, D, I}}, roadway::Roadway, vehicle_index::Int) where {D<:Union{VehicleDef, BicycleModel}, I}
```

Comment:
should be `D<:AbstractAgentDefinition` instead of the union

Reply:
ok, replaced all occurrences